core/types/src/attestation_key.rs

```diff
@@ -251,6 +253,91 @@ mod test {
         assert_eq!(key.algorithm_id().unwrap(), Algorithm::Reserved);
     }
 
+    #[test]
+    fn ct_eq_extended_attestation_key_id() {
+        let first = ExtendedAttestationKeyId::default();
+        let second = ExtendedAttestationKeyId::default();
+        let choice_result = first.ct_eq(&second);
+        let result: bool = From::from(choice_result);
+        assert!(result);
+    }
+
+    #[test]
+    fn ct_eq_version() {
+        let first = Version(2);
+        let second = Version(2);
+        let choice_result = first.ct_eq(&second);
+        let result: bool = From::from(choice_result);
+        assert!(result);
+    }
```

Comment:
:thought_balloon: I'm not sure tests at this level are very beneficial:
1. These are only testing `==` nothing is testing for `!=`.
2. They feel pretty noisy here and don't provide confidence on the full logic of the macro itself

Suggest making a `tests` directory in the macro crate and test the different types there.
This will:
- make it easier to test boundary conditions there
- make the tests live with the crate if it gets moved somewhere else
- prevents noise here. I know that we're testing the to/from logic. but we're not testing hash, ord, etc here.
```rust
use subtle::ConstantTimeEq;
use constant_time_derive::ConstantTimeEq;

#[derive(ConstantTimeEq)]
struct NamedFields {
    a: u8,
    b: u8,
}

#[test]
fn named_fields() {
    let a = NamedFields { a: 1, b: 2 };
    let b = NamedFields { a: 1, b: 2 };
    let c = NamedFields { a: 1, b: 3 };
    assert!(bool::from(a.ct_eq(&b)));
    assert!(bool::from(!a.ct_eq(&c)));
}

```